pineappl_py/src/grid.rs

```diff
@@ -383,15 +383,16 @@ impl PyGrid {
     ///     numpy.ndarray(float) :
     ///         cross sections for all bins, for each scale-variation tuple (first all bins, then
     ///         the scale variation)
+    #[pyo3(signature = (pdg_id, xfx, alphas, order_mask = None, bin_indices = None, lumi_mask= None, xi = None))]
     pub fn convolve_with_one<'py>(
         &self,
         pdg_id: i32,
         xfx: &Bound<'py, PyAny>,
         alphas: &Bound<'py, PyAny>,
-        order_mask: PyReadonlyArray1<bool>,
-        bin_indices: PyReadonlyArray1<usize>,
-        lumi_mask: PyReadonlyArray1<bool>,
-        xi: Vec<(f64, f64)>,
+        order_mask: Option<PyReadonlyArray1<bool>>,
```

Comment:
I noticed an inconsistency (?) that was introduced a long time ago, we use both `PyReadonlyArray1` and `Vec`. Does anyone happen to know the difference on the Python side?

Reply:
`PyReadonlyArray1` should come from the [numpy create](https://docs.rs/numpy/latest/numpy/borrow/type.PyReadonlyArray1.html) and thus should correspond to a numpy array - instead `Vec` is `std` and thus should correspond to a (builtin) `list`

Reply:
> `PyReadonlyArray1` should come from the [numpy create](https://docs.rs/numpy/latest/numpy/borrow/type.PyReadonlyArray1.html) and thus should correspond to a numpy array - instead `Vec` is `std` and thus should correspond to a (builtin) `list`

Yes, or also a `tuple`.

Reply:
So which one is more generic? If we change it to a `Vec`, would numpy arrays still be accepted?

Reply:
> So which one is more generic? If we change it to a Vec, would numpy arrays still be accepted?

`Vec<PyObject>` is more general than `PyReadonlyArray1<PyObject>`  as the former accepts all `list`, `tuple`, and `np.ndarray` while the latter can only accept `np.ndarray`. The exception is when `PyObject=bool`. This is because `bool` is not compatible with `np.bool_`.

As a specific exception, `Vec<(f64, f64)>` only accepts `tuple(float, float)` - which is what we want for `xi`.

Reply:
OK, thank you for looking this up! I presume it best we make as few changes as possible, so let's keep this in mind for the future.